astropy/timeseries/binned.py

```diff
@@ -0,0 +1,277 @@
+
+            # Now shift the array so that the first entry is 0
+            time_delta = np.roll(time_delta, 1)
+            time_delta[0] = 0. * u.s
+
+            # Make time_bin_start into an array
+            time_bin_start = time_bin_start + time_delta
+
+        else:
+
+            if len(self.colnames) > 0 and len(time_bin_start) != len(self):
+                raise ValueError("Length of 'time_bin_start' ({0}) should match "
+                                 "table length ({1})".format(len(time_bin_start), len(self)))
+
+            if time_bin_end is not None:
+                if time_bin_end.isscalar:
+                    times = time_bin_start.copy()
+                    times[:-1] = times[1:]
+                    times[-1] = time_bin_end
+                    time_bin_end = times
+                time_bin_size = (time_bin_end - time_bin_start).sec * u.s
+
+        self.add_column(time_bin_start, index=0, name='time_bin_start')
+        self.add_index('time_bin_start')
+
+        if time_bin_size.isscalar:
+            time_bin_size = np.repeat(time_bin_size, len(self))
+
+        self.add_column(time_bin_size, index=1, name='time_bin_size')
+
+    @property
+    def time_bin_start(self):
+        """
+        The start times of all the time bins.
+        """
+        return self['time_bin_start']
+
+    @property
+    def time_bin_center(self):
+        """
+        The center times of all the time bins.
+        """
+        return self['time_bin_start'] + self['time_bin_size'] * 0.5
+
+    @property
+    def time_bin_end(self):
+        """
+        The end times of all the time bins.
+        """
+        return self['time_bin_start'] + self['time_bin_size']
+
+    @property
+    def time_bin_size(self):
+        """
+        The sizes of all the time bins.
+        """
+        return self['time_bin_size']
+
+    def __getitem__(self, item):
+        if self._is_list_or_tuple_of_str(item):
+            if 'time_bin_start' not in item or 'time_bin_size' not in item:
+                out = QTable([self[x] for x in item],
+                             meta=deepcopy(self.meta),
+                             copy_indices=self._copy_indices)
+                out._groups = groups.TableGroups(out, indices=self.groups._indices,
+                                                 keys=self.groups._keys)
+                return out
+        return super().__getitem__(item)
+
+    @classmethod
+    def read(self, filename, time_bin_start_column=None, time_bin_end_column=None,
+             time_bin_size_column=None, time_bin_size_unit=None, time_format=None, time_scale=None,
+             format=None, *args, **kwargs):
+        """
+        Read and parse a file and returns a `astropy.timeseries.BinnedTimeSeries`.
+
+        This method uses the unified I/O infrastructure in Astropy which makes
+        it easy to define readers/writers for various classes
+        (http://docs.astropy.org/en/stable/io/unified.html). By default, this
```

Comment:
Use intersphinx instead of hardcoding URL?

Reply:
For docstrings I think the full URLs make sense as the :ref:s would be of no use to readers.